### PR TITLE
feat(api): add DELETE endpoint for host groups

### PR DESCRIPTION
**Pull request by abuathier**
## Description

Jira : MON-16029

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [X] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [X] 22.10.x
- [X] 23.04.x (master)

<h2> How this pull request can be tested ? </h2>

In jira ticket

#### Community contributors & Centreon team

- [X] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [X] I have commented my code, especially **hard-to-understand areas** of the PR.
- [X] I have **rebased** my development branch on the base branch (master, maintenance).

[MON-16029]: https://centreon.atlassian.net/browse/MON-16029?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: centreon/src/Core/HostGroup/Infrastructure/Repository/DbWriteHostGroupRepository.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+class DbWriteHostGroupRepository extends AbstractRepositoryDRB implements WriteHostGroupRepositoryInterface
+{
+    use LoggerTrait;
+
+    public function __construct(DatabaseConnection $db)
+    {
+        $this->db = $db;
+    }
+
+    public function deleteHostGroup(int $hostGroupId): void
```
Comment (jeremyjaouen) [inline centreon/src/Core/HostGroup/Infrastructure/Repository/DbWriteHostGroupRepository.php:46]: phpdoc

**Review by jeremyjaouen — APPROVED** [review]

**Review by TamazC — APPROVED** [review]

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -17259,3 +17259,9 @@ msgstr "Erreur durant la recherche de groupes d'hôtes"
 
 msgid "You are not allowed to access host groups"
 msgstr "Vous n'êtes pas autorisé à accéder aux groupes d'hôtes"
+
+msgid "Host group not found"
+msgstr "Groupe d'hôtes non trouvé"
+
+msgid "Error while deleting a host group"
+msgstr "Erreur durant la suppression d'un groupe d'hôte"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:17267]:
```suggestion
msgstr "Erreur durant la suppression d'un groupe d'hôtes"
```

File: centreon/src/Core/HostGroup/Application/UseCase/DeleteHostGroup/DeleteHostGroup.php [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    public function __invoke(int $hostGroupId, PresenterInterface $presenter): void
+    {
+        try {
+            if ($this->contact->isAdmin()) {
+                $presenter->setResponseStatus($this->deleteHostGroupAsAdmin($hostGroupId));
+            } elseif ($this->contactCanExecuteThisUseCase()) {
+                $presenter->setResponseStatus($this->deleteHostGroupAsContact($hostGroupId));
+            } else {
+                $this->error(
+                    "User doesn't have sufficient right to see host groups",
```
Comment (cg-tw) [inline centreon/src/Core/HostGroup/Application/UseCase/DeleteHostGroup/DeleteHostGroup.php:62]:
```suggestion
                    "User doesn't have sufficient rights to see host groups",
```

File: centreon/tests/php/Core/HostGroup/Application/UseCase/DeleteHostGroup/DeleteHostGroupTest.php [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+    ($this->useCase)($this->hostgroupId, $this->presenter);
+
+    expect($this->presenter->getResponseStatus())
+        ->toBeInstanceOf(ErrorResponse::class)
+        ->and($this->presenter->getResponseStatus()?->getMessage())
+        ->toBe(HostGroupException::errorWhileDeleting()->getMessage());
+});
+
+it('should present an ForbiddenResponse when the user has not the correct role', function (): void {
```
Comment (cg-tw) [inline centreon/tests/php/Core/HostGroup/Application/UseCase/DeleteHostGroup/DeleteHostGroupTest.php:88]:
```suggestion
it('should present a ForbiddenResponse when the user does not have the correct role', function (): void {
```

**Review by cg-tw — APPROVED** [review]

**Pull request merged**